prow/plugins/trigger/generic-comment.go

```diff
@@ -53,8 +73,18 @@ func handleGenericComment(c Client, trigger plugins.Trigger, gc github.GenericCo
 
 	refGetter := config.NewRefGetterForGitHubPullRequest(c.GitHubClient, org, repo, number)
 	presubmits := getPresubmits(c.Logger, c.GitClient, c.Config, org+"/"+repo, refGetter.BaseSHA, refGetter.HeadSHA)
+
+	pr, err := refGetter.PullRequest()
```

Comment:
Why do we need to do this so early? Calling it here will run it for every comment, If this call uses tokens, that scales very poorly.

Reply:
Thank you for the review. Fixed.